Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Generator.cs

```diff
@@ -24,36 +36,76 @@ public Generator()
+                var templateFinder = new CloudFormationTemplateFinder(_fileManager, _directoryManager);
+                var projectRootDirectory = string.Empty;
 
-                if (string.IsNullOrEmpty(projectRootDirectory))
-                    projectRootDirectory = templateFinder.DetermineProjectRootDirectory(lambdaMethod.SyntaxTree.FilePath);
-            }
+                foreach (var lambdaMethod in receiver.LambdaMethods)
+                {
+                    var lambdaMethodModel = semanticModelProvider.GetMethodSemanticModel(lambdaMethod);
+                    var model = LambdaFunctionModelBuilder.Build(lambdaMethodModel, configureMethodModel, context);
+
+                    // If there are more than one event, report them as errors
+                    if (model.LambdaMethod.Events.Count > 1)
+                    {
+                        foreach (var attribute in lambdaMethodModel.GetAttributes().Where(attribute => TypeFullNames.Events.Contains(attribute.AttributeClass.ToDisplayString())))
+                        {
+                            diagnosticReporter.Report(Diagnostic.Create(DiagnosticDescriptors.MultipleEventsNotSupported,
+                                Location.Create(attribute.ApplicationSyntaxReference.SyntaxTree, attribute.ApplicationSyntaxReference.Span),
+                                DiagnosticSeverity.Error));
+                        }
+
+                        // Skip multi-event lambda method from processing and check remaining lambda methods for diagnostics
+                        continue;
+                    }
 
-            annotationReport.CloudFormationTemplatePath = templateFinder.FindCloudFormationTemplate(projectRootDirectory);
-            var cloudFormationJsonWriter = new CloudFormationJsonWriter(new FileManager(), new JsonWriter());
-            cloudFormationJsonWriter.ApplyReport(annotationReport);
+                    var template = new LambdaFunctionTemplate(model);
+                    var sourceText = template.TransformText();
+                    context.AddSource($"{model.GeneratedMethod.ContainingType.Name}.g.cs", SourceText.From(sourceText, Encoding.UTF8, SourceHashAlgorithm.Sha256));
+
+                    // report every generated file to build output
+                    diagnosticReporter.Report(Diagnostic.Create(DiagnosticDescriptors.CodeGeneration, Location.None, $"{model.GeneratedMethod.ContainingType.Name}.g.cs", sourceText));
+
+                    annotationReport.LambdaFunctions.Add(model);
+
+                    if (string.IsNullOrEmpty(projectRootDirectory))
+                        projectRootDirectory = templateFinder.DetermineProjectRootDirectory(lambdaMethod.SyntaxTree.FilePath);
+                }
+
+                annotationReport.CloudFormationTemplatePath = templateFinder.FindCloudFormationTemplate(projectRootDirectory);
+                var cloudFormationJsonWriter = new CloudFormationJsonWriter(_fileManager, _jsonWriter, diagnosticReporter);
```

Comment:
The `diagnosticReporter` only needs the content of the `serveless.template` and can be used directly after the `cloudFormationJsonWriter.ApplyReport(annotationReport)` invocation.

Reply:
Discussed offline.